lasagne/layers/special.py

```diff
@@ -662,38 +688,46 @@ def get_output_for(self, inputs, **kwargs):
         # see eq. (1) and sec 3.1 in [1]
         # Get input and destination control points
         input, dest_offsets = inputs
-        return _transform_thin_thin_plate_spline(dest_offsets,
-                                                 input,
-                                                 self.source_points,
-                                                 self.L_inv,
-                                                 self.downsample_factor)
+        return _transform_thin_thin_plate_spline(
+                dest_offsets, input, self.right_mat, self.L_inv,
+                self.source_points, self.out_height, self.out_width,
+                self.precompute_grid, self.downsample_factor)
```

Comment:
Good catch on both! I removed the extra `thin` and added in a test for `precompute_grid='auto'` with a variable input shape.